### PR TITLE
[13.0][mig][purchase_manual_delivery]

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
Standard migration.

cc @AaronHForgeFlow @LoisRForgeFlow @AdriaGForgeFlow

### CODE REVIEW

**Comment by CasVissers-360ERP** [comment]
@JordiBForgeFlow 
My main concern with this module is that pending_to_receive quantities are not visible in other parts of Odoo (i.e. virtual qty).
 
Would it be an option to add this to _compute_quantities_dict are do you think is it too intrusive? Which I think it is actually. Just wanted to raise to concern.

**Comment by JordiBForgeFlow** [comment]
@CasVissers-360ERP I agree, but at the same time virtual quantity is really a terrible idea in the context where you have a long delivery schedule. What good does it make to say that the virtual quantity is 10000 if you are pending to receive this quantity over 1 year?

But that's how odoo works, and perhaps you are right that we should include it in _compute_quantities_dict. Will do that.

File: purchase_manual_delivery/views/purchase_order_views.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
```
Comment (CasVissers-360ERP) [inline purchase_manual_delivery/views/purchase_order_views.xml:2]: XML in this file is not easy to read, can this be improved?

File: purchase_manual_delivery/wizard/create_manual_stock_picking.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
```
Comment (CasVissers-360ERP) [inline purchase_manual_delivery/wizard/create_manual_stock_picking.xml]: XML in this file is not easy to read, can this be improved?

File: purchase_manual_delivery/views/purchase_order_views.xml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                <attribute name="invisible">1</attribute>
+            </button>
+            <button id="draft_confirm" position="after">
+                <button
+                    name="button_confirm_manual"
+                    type="object"
+                    states="draft"
+                    string="Confirm Order"
+                    id="draft_confirm_manual"
+                />
```
Comment (CasVissers-360ERP) [inline purchase_manual_delivery/views/purchase_order_views.xml:28]: Don't understand the need to create a separate method and button for this. Why not just update the context in button_confirm directly.

File: purchase_manual_delivery/wizard/create_manual_stock_picking.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+                    "date_planned": line.date_planned,
+                    "price_unit": line.price_unit,
+                    "product_qty": line.product_qty,
+                    "existing_qty": line.existing_qty,
+                    "remaining_qty": line.product_qty - line.existing_qty,
+                    "qty": line.product_qty - line.existing_qty,
+                    "product_uom": line.product_uom.id,
+                    "currency_id": line.currency_id.id,
+                    "partner_id": line.partner_id.id,
+                    # 'taxes_id': line.taxes_id.ids,
```
Comment (CasVissers-360ERP) [inline purchase_manual_delivery/wizard/create_manual_stock_picking.py:83]: Can be removed?

File: purchase_manual_delivery/models/purchase_order.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    existing_qty = fields.Float(
+        compute="_compute_existing_qty",
+        string="Existing Qty",
+        compute_sudo=True,
+        digits=dp.get_precision("Product Unit of Measure"),
```
Comment (CasVissers-360ERP) [inline purchase_manual_delivery/models/purchase_order.py]: digits="Product Unit of Measure",

File: purchase_manual_delivery/models/purchase_order.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            return
+        return super()._create_picking()
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    existing_qty = fields.Float(
+        compute="_compute_existing_qty",
+        string="Existing Qty",
```
Comment (CasVissers-360ERP) [inline purchase_manual_delivery/models/purchase_order.py]: string="Receiving" to keep in line with schedule lines naming

File: purchase_manual_delivery/models/purchase_order.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        compute_sudo=True,
+        digits=dp.get_precision("Product Unit of Measure"),
+        help="Quantity already planned or shipped " "(stock movements already created)",
+    )
+    pending_to_receive = fields.Boolean(
+        compute="_compute_existing_qty",
+        compute_sudo=True,
+        store=True,
+        string="Pending Qty to Receive",
```
Comment (CasVissers-360ERP) [inline purchase_manual_delivery/models/purchase_order.py]: string="Pending to Receive" to keep in line with schedule lines naming

File: purchase_manual_delivery/models/purchase_order.py (hunk 5 above)
Comment (CasVissers-360ERP) [reply purchase_manual_delivery/models/purchase_order.py]: Still open, after refactoring.

File: purchase_manual_delivery/wizard/create_manual_stock_picking.py (hunk 4 above)
Comment (CasVissers-360ERP) [reply purchase_manual_delivery/wizard/create_manual_stock_picking.py:83]: Feedback still valid in my opinion.

File: purchase_manual_delivery/wizard/create_manual_stock_picking.xml (hunk 2 above)
Comment (CasVissers-360ERP) [reply purchase_manual_delivery/wizard/create_manual_stock_picking.xml]: Feedback still valid in my opinion.

File: purchase_manual_delivery/views/purchase_order_views.xml (hunk 3 above)
Comment (CasVissers-360ERP) [reply purchase_manual_delivery/views/purchase_order_views.xml:28]: Non-important feedback. Still valid maybe though.

File: purchase_manual_delivery/models/purchase_order.py (hunk 7 above)
Comment (CasVissers-360ERP) [reply purchase_manual_delivery/models/purchase_order.py]: Still open, after refactoring.

**Comment by HviorForgeFlow** [comment]
Hello @CasVissers-360ERP can you take a look on new incoming changes? Thanks!

**Review by lfreeke — APPROVED** [review]
:+1: Functional test

**Review by HviorForgeFlow — APPROVED** [review]

**Pull request merged**